packages/error-reporting/src/google-apis/auth-client.js

```diff
@@ -88,26 +88,33 @@ class RequestHandler extends common.Service {
    */
   constructor(config, logger) {
     var pid = config.getProjectId();
+    // If an API key is provided, do not try to authenticate.
+    var tryAuthenticate = !config.getKey();
     super({
       packageJson: pkg,
-      baseUrl: 'https://clouderrorreporting.googleapis.com/v1beta1/',
+      baseUrl: API,
       scopes: SCOPES,
       projectId: pid !== null ? pid : undefined,
-      projectIdRequired: true
+      projectIdRequired: true,
+      customEndpoint: !tryAuthenticate
     }, config);
     this._config = config;
     this._logger = logger;
 
     var that = this;
-    this.authClient.getToken(function(err, token) {
-      if (err) {
-        that._logger.error([
-          'Unable to find credential information on instance. This library',
-          'will be unable to communicate with the Stackdriver API to save',
-          'errors.  Message: ' + err.message
-        ].join(' '));
-      }
-    });
+    if (tryAuthenticate) {
```

Comment:
It might be possible to access the [list](https://cloud.google.com/error-reporting/reference/) of reported errors as a way to notify the user if the key is invalid as soon as possible (as apposed to having the user not know the API key is invalid until an error is actually reported).